group_12/sharpen.py

```diff
@@ -0,0 +1,36 @@
+import numpy as np
+import pandas as pd
+# import os
+from scipy.signal import convolve2d
+from skimage.color import rgb2gray
+
+import matplotlib.pyplot as plt
+# %matplotlib inline
+
+def read_image(filename):
+    return plt.imread(filename)
+
+def sharpen(img):
+    """
+    Detects and enhances the edges in the image and
+    returns a sharpened and monotoned version (the same size as the original).
+
+    Parameters
+    ----------
+    img : a (H,W,3) numpy array
+      the image to be processed
+
+    Returns
+    --------
+    img : a (H,W) numpy array, returns the sharpened and monotoned image
+
+    """
+    img2 = rgb2gray(img)
+    flt = -np.ones((3,3))/22
+    flt[1,1] = 1
```

Comment:
May add some short comments to explain what is the transforming function doing. I would be better if some error handling tests can be added to the function to check input values/types.

Reply:
That's a perfect suggestion.  I will do that.